### PR TITLE
[Bug]Goals: Applying templates would zero non-templated categories

### PR DESCRIPTION
**Pull request by shall0pass**
Using the Apply Template or Overwrite option would result in non-templated categories being overwritten. First reported : https://discord.com/channels/937901803608096828/1137329898999001140/1137329898999001140

### CODE REVIEW

**Review by MatissJanis — APPROVED** [review]
Rubber stamp approval as I don't really use the goal schedules feature, so not sure what this does exactly. But trusting you.

File: packages/loot-core/src/server/budget/goaltemplates.ts [hunk 1]
```diff
@@ -97,22 +91,27 @@ async function processTemplate(month, force, category_templates) {
...
-  if (isReflectBudget()) {
-    for (let l = 0; l < originalCategoryBalance.length; l++) {
-      if (originalCategoryBalance[l].isIncome) {
-        await setBudget({
-          category: originalCategoryBalance[l].cat.id,
-          month,
-          amount: originalCategoryBalance[l].amount,
-        });
-      }
+    if (budgeted) {
```
Comment (shall0pass) [inline packages/loot-core/src/server/budget/goaltemplates.ts:94]: This is creating a list of of all categories in the budget along with the currently filled in amount and whether or not it has a template.

File: packages/loot-core/src/server/budget/goaltemplates.ts [hunk 2]
```diff
@@ -97,22 +91,27 @@ async function processTemplate(month, force, category_templates) {
...
+      originalCategoryBalance.push({
+        category: category.id,
+        amount: budgeted,
+        isIncome: category.is_income,
+        isTemplate: template ? true : false,
+      });
+      setToZero.push({
```
Comment (shall0pass) [inline packages/loot-core/src/server/budget/goaltemplates.ts:101]: This is doing the same thing, but substituting the amount for 0 to only zero.  We're also recording whether or not it's an income category.

File: packages/loot-core/src/server/budget/goaltemplates.ts [hunk 3]
```diff
@@ -97,22 +91,27 @@ async function processTemplate(month, force, category_templates) {
...
+        category: category.id,
+        amount: 0,
+        isIncome: category.is_income,
+        isTemplate: template ? true : false,
+      });
     }
   }
+  await setGoalBudget({
+    month,
+    templateBudget: setToZero.filter(
```
Comment (shall0pass) [inline packages/loot-core/src/server/budget/goaltemplates.ts:111]: This filters out categories that don't have templates and income categories.

File: packages/loot-core/src/server/budget/goaltemplates.ts [hunk 4]
```diff
@@ -231,14 +230,20 @@ async function processTemplate(month, force, category_templates) {
...
-        category: originalCategoryBalance[l].cat.id,
+        category: originalCategoryBalance[l].category,
         month,
         amount: originalCategoryBalance[l].amount,
       });
       //if overwrite is not preferred, remove template errors for category
       let j = errors.length;
       for (let k = 0; k < j; k++) {
-        if (errors[k].includes(originalCategoryBalance[l].cat.name)) {
+        if (
```
Comment (shall0pass) [inline packages/loot-core/src/server/budget/goaltemplates.ts:240]: In the case overwrite is preferred, this filters out categories that don't have templates and income categories and applies the originally budgeted amount.

File: packages/loot-core/src/server/budget/goaltemplates.ts [hunk 5]
```diff
@@ -231,14 +230,20 @@ async function processTemplate(month, force, category_templates) {
     //if overwrite is not preferred, set cell to original value
     for (let l = 0; l < originalCategoryBalance.length; l++) {
       await setBudget({
```
Comment (shall0pass) [inline packages/loot-core/src/server/budget/goaltemplates.ts:232]: Correction:  This is where it is set to the original balance, but only if it has a template, had a budgeted value, and is not income.

**Comment by shall0pass** [comment]
There 500 error in the Test/migrations test pointing to the URL: https://nodejs.org/dist/v19.9.0/node-v19.9.0-linux-x64.tar.gz
I'm assuming that's not on our end......

**Review by MatissJanis — APPROVED** [review]
Thanks for the comments

**Pull request merged**